haystack/nodes/base.py

```diff
@@ -74,6 +70,38 @@ def __init_subclass__(cls, **kwargs):
         # Enables generic load() for all specific component implementations.
         cls._subclasses[cls.__name__] = cls
 
+    @property
+    def name(self) -> Optional[str]:
+        if "name" not in self._component_config:
+            return None
+        return self._component_config["name"]
+
+    @name.setter
+    def name(self, value: str):
+        self._component_config["name"] = value
+
+    @property
+    def sub_components(self) -> List[BaseComponent]:
```

Comment:
The name might be slightly confusing... This is the list of nodes this components depends on, right? We might call it `required_components` or something that highlights the dependency relationship

Reply:
I rather go with `utilized_components`. At least that makes the most sense to me ;-)